### ISSUE TITLE
acceptance: fail to complete initial replication

### ISSUE THREAD

**Issue by tschottdorf** [cockroachdb/cockroach#7113]
See `TestStatusServer` in https://circleci.com/gh/tschottdorf/cockroach/2278#artifacts

```
	testing.go:113: acceptance/replication_test.go:72: condition failed to evaluate within 15s: expected 3 replicas, only found 2
```

It looks like the first range didn't replicate all of the ranges successfully (though there aren't any obvious errors). Might be related to #6878.

**Comment by tamird** [comment]
Not sure what's going on here.

roach0 proposes replication of ranges 3 and 4 to roach2
roach1 proposes replication of range 2 to roach2
roach2 only receives snapshots for ranges 2 and 4

also, roach0's logs are full of warnings like these for ranges 3 and 5 (the indexes never change):
```
W160608 19:35:26.758968 storage/raft_log_queue.go:118  storage/raft_log_queue.go:103: raft log's oldest index (0) is less than the first index (11) for range 5
W160608 19:35:27.360272 storage/raft_log_queue.go:118  storage/raft_log_queue.go:103: raft log's oldest index (0) is less than the first index (26) for range 3
```

**Comment by bdarnell** [comment]
Several snapshots for range 4 failed (silently) and had to be retried.

On roach0:
```
I160608 19:35:25.670893 storage/replica_raftstorage.go:552  generated snapshot for range 4 at index 21 in 180.289µs. encoded size=5243, 44 KV pairs, 3 log entries
I160608 19:35:25.687506 storage/replica.go:1376  raft: proposing ADD_REPLICA {NodeID:2 StoreID:2 ReplicaID:2} for range 4
I160608 19:35:25.753352 storage/replica_raftstorage.go:552  generated snapshot for range 4 at index 25 in 311.204µs. encoded size=7702, 46 KV pairs, 7 log entries
I160608 19:35:26.675745 storage/replica_raftstorage.go:552  generated snapshot for range 4 at index 33 in 308.784µs. encoded size=7781, 65 KV pairs, 3 log entries
I160608 19:35:26.687900 storage/replica.go:1376  raft: proposing ADD_REPLICA {NodeID:3 StoreID:3 ReplicaID:3} for range 4
I160608 19:35:26.698072 storage/replica_raftstorage.go:552  generated snapshot for range 4 at index 37 in 312.025µs. encoded size=10356, 67 KV pairs, 7 log entries
I160608 19:35:26.704523 storage/replica_raftstorage.go:552  generated snapshot for range 4 at index 39 in 308.141µs. encoded size=10635, 66 KV pairs, 9 log entries
```

roach1 applied a snapshot at index 25; roach2 applied a snapshot at index 37. The others are redundant. The one at index 21 must have failed and been discarded (or reordered relative to the later ones?). The others were either generated redundantly (which is something we should try to avoid) or also experienced unknown failures.

**Comment by tamird** [comment]
Could be related to snapshot corruption fixed in #7299; closing until it resurfaces.
